src/js/keyboardInput.js

```diff
@@ -206,14 +216,18 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
                 }
             }
             if (ch !== "") {
-                input.val(ch);
-                // programmatic change of the input value does not
-                // fire a change event, so we trigger it explicitly
-                input.triggerHandler("change");
+                that.events.keypress.fire(ch);
             }
         });
     };
 
+    gpii.firstDiscovery.keyboardInput.setElementValueAndTriggerChange = function (elem, value) {
```

Comment:
We have captured this work as a JIRA issue: http://issues.fluidproject.org/browse/FLUID-5666